napari_plugin_manager/_tests/test_qt_plugin_dialog.py

```diff
@@ -166,42 +188,20 @@ def set_blocked(self, plugin, blocked):
 
     widget = qt_plugin_dialog.QtPluginDialog()
     widget.show()
-    qtbot.wait(300)
-    qtbot.add_widget(widget)
-    yield widget
-    widget.hide()
-    widget._add_items_timer.stop()
-    assert not widget._add_items_timer.isActive()
+    qtbot.waitUntil(widget.isVisible, timeout=300)
 
+    def available_list_populated():
+        return widget.available_list.count() == N_MOCKED_PLUGINS
 
-@pytest.fixture
-def plugin_dialog_constructor(qtbot, monkeypatch):
-    """
-    Fixture that provides a plugin dialog for a constructor based install.
-    """
-    monkeypatch.setattr(
-        qt_plugin_dialog,
-        "iter_napari_plugin_info",
-        _iter_napari_pypi_plugin_info,
-    )
-
-    # This is patching `napari.utils.misc.running_as_constructor_app` function
-    # to mock a constructor based install.
-    monkeypatch.setattr(
-        qt_plugin_dialog,
-        "running_as_constructor_app",
-        lambda: True,
-    )
-    widget = qt_plugin_dialog.QtPluginDialog()
-    widget.show()
-    qtbot.wait(300)
+    qtbot.waitUntil(available_list_populated, timeout=3000)
     qtbot.add_widget(widget)
     yield widget
     widget.hide()
     widget._add_items_timer.stop()
+    assert not widget._add_items_timer.isActive()
 
 
-def test_filter_not_available_plugins(plugin_dialog_constructor):
+def test_filter_not_available_plugins(qtbot, plugin_dialog):
```

Comment:
```suggestion
def test_filter_not_available_plugins(plugin_dialog):
```

I do not see qtbot usage in this test